indra/newview/llpuppetmotion.cpp

```diff
@@ -991,7 +989,6 @@ void LLPuppetMotion::applyBroadcastEvent(const LLPuppetJointEvent& event, Timest
         {
             joint_state->setScale(event.getScale());
         }
-        joint_state->setUsage((U32)(flags & LLIK::MASK_JOINT_STATE_USAGE));
```

Comment:
This is the actual fix: we need to LLJointState::setUsage() BEFORE trying to update those fields.

I think the only reason this didn't ALWAYS crash was because we were not clearing the Usage flags elsewhere, so once the flags were set they stayed set for future joint updates.

Note: this is a theoretical fix since I haven't been able to reproduce the problem, but I'm confident this the fix: this is the only place I could find where we were trying to update the properties before declaring the usage.